### PR TITLE
Asymmetrical env change

### PR DESCRIPTION
**Pull request by swom**

### CODE REVIEW

File: simulation.cpp [hunk 1]
```diff
@@ -649,5 +673,44 @@ void test_simulation() noexcept//!OCLINT test may be many
     }
 #endif
 
+#define FIX_ISSUE_249
+#ifdef FIX_ISSUE_249
```
Comment (Clemcarle) [inline simulation.cpp:677]:
```suggestion
#ifdef FIX_ISSUE_249
///Simulation changes environment/function according to the specific change frequency of each function
```

File: simulation.h [hunk 2]
```diff
@@ -97,17 +105,23 @@ class simulation
     ///Returns the number of generatiosn for which the simualtion has to run
     const int& get_n_gen() const noexcept {return m_n_generations;}
 
-    ///returns const ref to
-    const std::bernoulli_distribution& get_t_change_env_distr() const noexcept {return m_t_change_env_distr;}
-    std::bernoulli_distribution& get_t_change_env_distr() noexcept {return m_t_change_env_distr;}
+    ///returns const ref to Bernoulli distribution for change freq of A
+    const std::bernoulli_distribution& get_t_change_env_distr_A() const noexcept {return m_t_change_env_distr_A;}
+
+    ///returns const ref to Bernoulli distribution for change freq of A
```
Comment (Clemcarle) [inline simulation.h:111]:
```suggestion
    ///returns const ref to Bernoulli distribution for change freq of B
```

**Review by Clemcarle — CHANGES_REQUESTED** [review]

File: environment.h [hunk 3]
```diff
@@ -133,6 +135,11 @@ void switch_env_function(environment &e);
 
 }
 
+
+///homebrew version of json loading and saving to load and save cue distributions for environment
```
[the remark this answers is not in the document]
Comment (swom) [reply environment.h:139]: yup, the test in simulation cpp line 400 something takes care of it I think

**Review by Clemcarle — APPROVED** [review]

**Comment by Clemcarle** [comment]
I'll let you merge because I don't know if you're still working on something

**Pull request merged**